### PR TITLE
Ellipsis in subscript

### PR DESCRIPTION
**Pull request by Krovatkin**

### CODE REVIEW

**Comment by Krovatkin** [comment]
https://github.com/pytorch/pytorch/issues/17389 

tagging @zdevito for a quick progress update: I had very little time to work on this last week and most of this week. I should move faster given that my bootcamp journey is over now.

File: torch/csrc/jit/script/compiler.cpp [hunk 1]
```diff
@@ -2408,15 +2402,35 @@ struct to_ir {
       dim++;
     };
 
+    // before ellipsis, dimension index should be `dim`
```
Comment (suo) [inline torch/csrc/jit/script/compiler.cpp]: the logic here is a little obscure. It's hard to track what's going on in an out of the loop, various lambdas, etc. Also, `handle_dim` is not very descriptive. Is there a way to restructure this so it's more clear for readers?

Comment (Krovatkin) [reply torch/csrc/jit/script/compiler.cpp]: I looked more into whether we could untangle this and there doesn't seem to be a lot we could do 😢 I added quite a few comments to explain why the slicing logic structured this particular way.

Comment (Krovatkin) [reply torch/csrc/jit/script/compiler.cpp]: Also renamed some variables in a hope to make it a bit clearer

**Review by suo — APPROVED** [review]
this looks good, but try rebasing past the test failures before landing

**Pull request closed without merging**